pandas/types/common.py

```diff
@@ -491,3 +491,27 @@ def pandas_dtype(dtype):
         return dtype
 
     return np.dtype(dtype)
+
+
+def _is_fillable_value(value):
+    pandas_ts_types = ('Timestamp', 'Period', 'Timedelta')
+    pandas_block_types = ('Series', 'DataFrame')
+
+    if any([isinstance(value, (list, dict)),
+            callable(value),
+            (not (isinstance(value, string_types) or
+                  isinstance(value, (int, float, complex, str, None.__class__)) or
+                  is_numeric_dtype(value) or
+                  is_datetime_or_timedelta_dtype(value) or
```

Comment:
Well, I've been following the `fillna` behavior thus far. Right now `fillna` would convert that input to a timestamp in `ns`. Same with `float` or `bool`. And upcast the column to `object` dtype to fit a `str` fill.

`fillna` behaves that way AFAIK because it's convenient to propagate a `0` or an `np.nan` or whatever other out-of-type sentinel value across an entire `DataFrame` all at once, instead of having to go column-by-column.

The same argument might apply for `fill_value`, but, I do see it being a far weaker one. So if you think that it's OK for `fill_value` to have a separate, stricter behavior than `fillna`, sure.